File: cpp/estrutura_sequencial/retangulo.cpp
/* Problema "retangulo"
 Fazer um programa para ler as medidas da base e altura de um retângulo.
 Em seguida, mostrar o valor da área, perímetro e diagonal deste retângulo, com quatro casas decimais,
*/

#include <cmath>
#include <iomanip>
#include <iostream>

using namespace std;

int main()
{
  double base, height, area, perimeter, diagonal;

  cout << "Base do triângulo: ";
  cin >> base;

  cout << "Altura do triângulo: ";
  cin >> height;

  area = height * base;
  perimeter = 2 * (base + height);
  diagonal = sqrt(base * base + height * height);

  cout << fixed << setprecision(4);
  cout << "ÁREA = " << area << endl;
  cout << "PERÍMETRO = " << perimeter << endl;
  cout << "DIAGONAL = " << diagonal << endl;

  return 0;
}

File: csharp/Source/Estruturas/Sequencial/Retangulo.cs
namespace Source.Estruturas.Sequencial;

/**
 * Problema "retangulo"
 * Fazer um programa para ler as medidas da base e altura de um retângulo.
 * Em seguida, mostrar o valor da área, perímetro e diagonal deste retângulo,
 * com quatro casas decimais,
 */

internal static class Retangulo
{
    static public void Execute()
    {
        Console.Write("Base do triângulo: ");
        double baseLength = double.Parse(Console.ReadLine()!);

        Console.Write("Altura do triângulo: ");
        double height = double.Parse(Console.ReadLine()!);

        double area = height * baseLength;
        double perimeter = 2 * (baseLength + height);
        double diagonal = Math.Sqrt(Math.Pow(baseLength, 2.0) + height * height);

        Console.WriteLine("ÁREA = {0:F4}\n", area);
        Console.WriteLine("PERÍMETRO = {0:F4}\n", perimeter);
        Console.WriteLine("DIAGONAL = {0:F4}\n", diagonal);
    }
}